performance/README.md

```diff
@@ -0,0 +1,16 @@
+# Performance Regression Testing
+This directory includes dbt project setups to test on and a test runner written in Rust which runs specific dbt commands on each of the projects. Orchestration is done via the GitHub Action workflow in `/.github/workflows/performance.yml`. The workflow is scheduled to run every night, but it can also be triggered manually.
+
+The github workflow hardcodes our baseline branch for performance metrics as `0.20.latest`. As future versions become faster, this branch will be updated to hold us to those new standards.
+
+## Adding a new dbt project
+Just make a new directory under `performance/projects/`. It will automatically be picked up by the tests.
```

Comment:
I'm wondering if we really want to check in the thousands of files associated with the performance tests? Could we put the projects we want to test into separate repos?

Reply:
yes. we absolutely could do that. The advantage would be that we don't have to download all those files when we checkout dbt, but it's also another repository to manage. I just measured that a dbt project with 2k files is roughly 6.3k to download which I don't consider a deal breaker. But I'm honestly open to both approaches.

Reply:
I like that this offers general support for performance-testing any project. I'm thinking about specific projects that we've known to parse slowly, and how great it would be to stress-test a putative improvement in a more-rigorous manner.

As far as running this nightly: If we're going to use generated projects, I'd be open to including the [project-generation code](https://github.com/dbt-labs/dbt-timing-project) in this repo, and running the project generation as a step in the `performance` workflow.

Reply:
@jtcohen6 that is a good next step, however most places that do this kind of performance work (and publish it) usually check in projects to prove the metrics aren't changing because the project is changing. We could go against the grain on that one, but I'm inclined to check it in if it doesn't cause any tangible issues.

Reply:
Ok cool, I buy that (and remember you having told me this). I'm all for doing this in a way that follows existing practice.